WordPress/src/main/java/org/wordpress/android/ui/themes/ThemeBrowserActivity.java

```diff
@@ -462,59 +496,33 @@ private void startWebActivity(String themeId, ThemeWebActivity.ThemeWebActivityT
         ToastUtils.showToast(this, toastText, ToastUtils.Duration.SHORT);
     }
 
-    @Override
-    public void onActivateSelected(String themeId) {
-        activateTheme(themeId);
-    }
-
-    @Override
-    public void onTryAndCustomizeSelected(String themeId) {
-        startWebActivity(themeId, ThemeWebActivity.ThemeWebActivityType.PREVIEW);
-    }
-
-    @Override
-    public void onViewSelected(String themeId) {
-        startWebActivity(themeId, ThemeWebActivity.ThemeWebActivityType.DEMO);
-    }
-
-    @Override
-    public void onDetailsSelected(String themeId) {
-        startWebActivity(themeId, ThemeWebActivity.ThemeWebActivityType.DETAILS);
-    }
-
-    @Override
-    public void onSupportSelected(String themeId) {
-        startWebActivity(themeId, ThemeWebActivity.ThemeWebActivityType.SUPPORT);
-    }
-
-    @Override
-    public void onSearchClicked() {
-        mIsInSearchMode = true;
-        AnalyticsUtils.trackWithSiteDetails(AnalyticsTracker.Stat.THEMES_ACCESSED_SEARCH, mSite);
-        addSearchFragment();
-    }
-
-    public class FetchThemesTask extends AsyncTask<JSONObject, Void, ArrayList<Theme>> {
+    /**
+     * Converts themes from a JSON REST response object to a list of {@link Theme}'s and saves them to local database.
+     * A {@link JSONObject} containing a {@link JSONArray} with key "themes" is expected.
+     */
+    private class DeserializeThemesRestResponse extends AsyncTask<JSONObject, Void, ArrayList<Theme>> {
         @Override
         protected ArrayList<Theme> doInBackground(JSONObject... args) {
-            JSONObject response = args[0];
-            final ArrayList<Theme> themes = new ArrayList<>();
+            if (args[0] == null) {
+                return null;
+            }
+
+            // JSONArray of themes is required
+            final JSONArray array = args[0].optJSONArray("themes");
+            if (array == null) {
+                return null;
+            }
 
-            if (response != null) {
-                JSONArray array;
+            // convert JSON themes to Theme models and store them in database
+            final ArrayList<Theme> themes = new ArrayList<>();
+            final int count = array.length();
```

Comment:
Since array's length is not being altered, I don't think we need to introduce this variable. I might be wrong about this, but when I see something like this, I'll always assume that the array is being somehow altered. It might be a personal preference though.